src/autoprops.rs

```diff
@@ -0,0 +1,260 @@
+                    syn::FnArg::Receiver(receiver) => Some(receiver),
+                    _ => None,
+                })
+                .collect::<Vec<_>>();
+            let args = sig
+                .inputs
+                .iter()
+                .filter_map(|arg| match arg {
+                    syn::FnArg::Typed(syn::PatType { pat, .. }) => Some(quote! { #pat, }),
+                    _ => None,
+                })
+                .collect::<Vec<_>>();
+            let phantom = sig.generics.type_params().next().is_some().then(|| {
+                quote! {
+                    phantom: _,
+                }
+            });
+            quote! {
+                #(#receivers,)* #properties_name {
+                    #(#args)*
+                    #phantom
+                }: &#properties_name #type_generics
+            }
+        } else {
+            quote! {}
+        };
+        let clones = sig
+            .inputs
+            .iter()
+            .filter_map(|arg| match arg {
+                syn::FnArg::Typed(syn::PatType { pat, ty, .. })
+                    if !matches!(**ty, syn::Type::Reference(_)) =>
+                {
+                    Some(quote! {
+                        let #pat = ::yew::html::ImplicitClone::implicit_clone(#pat);
+                    })
+                }
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+        let output = &sig.output;
+        let impl_generics_with_defaults = &sig.generics;
+
+        quote! {
+            #(#attrs)*
+            #vis fn #fn_name #impl_generics_with_defaults (#inputs) #output #where_clause {
+                #(#clones)*
+                #block
+            }
+        }
+    }
+
+    fn print_properties_struct(&self) -> proc_macro2::TokenStream {
+        let properties_name = &self.properties_name;
+        let syn::ItemFn { vis, sig, .. } = &self.item_fn;
+
+        if !self.needs_a_properties_struct() {
+            return quote! {};
+        }
+
+        let (impl_generics, type_generics, where_clause) = sig.generics.split_for_impl();
+        let fields = sig
+            .inputs
+            .iter()
+            .filter_map(|arg| match arg {
+                syn::FnArg::Typed(syn::PatType { attrs, pat, ty, .. }) => match ty.as_ref() {
+                    syn::Type::Reference(syn::TypeReference { elem, .. }) => {
+                        Some(quote! { #(#attrs)* #vis #pat: #elem, })
+                    }
+                    _ => Some(quote! { #(#attrs)* #vis #pat: #ty, }),
+                },
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+        let type_params = sig
+            .generics
+            .type_params()
+            .map(|param| &param.ident)
+            .collect::<Vec<_>>();
+        let phantom = (!type_params.is_empty()).then(|| {
+            quote! {
+                #[prop_or_default]
+                #vis phantom: ::std::marker::PhantomData <( #(#type_params),* )>,
```

Comment:
```suggestion
                #vis __yew_auto_phantom: ::std::marker::PhantomData <( #(#type_params),* )>,
```

This type should be prefixed to minimise chance of collision.

Reply:
Done in e323bee